### ISSUE TITLE
`as_factor.labelled` should preserve the variable label

### ISSUE THREAD

**Issue by LaDilettante** [hadley/haven#177]
When convert a `labelled` vector to a factor using `as_factor`, the variable name, stored in the attribute `label`, should be preserved. I'd be happy to contribute a pull request if you deem this a good idea.

Example:

```
# Create a labelled vector with both (variable) label and (value) labels
my_var <- sample(c(0, 1), 10, replace = TRUE)
class(my_var) <- "labelled"
attr(my_var, "label") <- "My Variable"
attr(my_var, "labels") <- c("Agree" = 1, "Disagree" = 0)


my_var # Has both variable label and value labels
as_factor(my_var) # No longer has the variable label
```

**Comment by hadley** [comment]
Yes, I'd be happy to review a pull request

**Comment by hadley** [comment]
Fixed by #179
